x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.ts

```diff
@@ -28,8 +28,10 @@ export type MapFilters = {
   searchSessionId?: string;
   timeFilters: TimeRange;
   timeslice?: Timeslice;
-  zoom: number;
-  isReadOnly: boolean;
+  zoom?: number;
```

Comment:
Why did you make zoom and isReadOnly optional?

Reply:
we were not saving `zoom` and `isreadonly` on the `next` and `previousQueryContext`. This PR annotates these variables to be more explicit.  https://github.com/elastic/kibana/pull/104691/files#diff-a21e65b32e4fae1f3d00edeb4e1de3a8ea38bb06327bf3911c08aa2536d7e0bbR269

If we keep the required-typing, then we'd also need to track those properties on these objects. We probably can. Or we should introduce another type specifically for these types.

thoughts?

Reply:
I would not pollute DataFilters by making zoom and isreadonly optional because of overloading the type to make it work for typing next and previousQueryContext. I think next and PreviousQueryContext do not need a strict type anyway since they are only used for a comparison in the scope of a single method.